client/src/components/ProtectedRoutes/ProtectedRoutes.tsx

```diff
@@ -1,18 +1,18 @@
-import React from 'react';
-import { Route, Redirect, RouteProps } from 'react-router-dom';
+import { Route, RouteProps, RouteComponentProps } from 'react-router-dom';
 import { useAuth } from '../../context/useAuthContext';
 
-interface Props extends RouteProps {
-  component: React.ComponentType;
-}
-
-const ProtectedRoute = ({ component: Component, ...rest }: Props): JSX.Element => {
+const ProtectedRoute = ({ children, component: Component, ...routeProps }: RouteProps): JSX.Element => {
   const { loggedInUser } = useAuth();
   return (
     <Route
-      render={(props) => (loggedInUser ? <Component {...rest} {...props} /> : <Redirect to="/login" />)}
-      {...rest}
+      {...routeProps}
+      render={({ location, ...restProps }: RouteComponentProps) =>
+        loggedInUser && children
```

Comment:
If users access private routes, wouldn't we want them to be redirected to login? :)

Reply:
yeah, I was thinking in wrong way, and I changed back to previous code, but make a little change